### ISSUE TITLE
simplecpp:realFileName returns the wrong root path

### ISSUE THREAD

**Issue by amai2012** [danmar/simplecpp#45]
User _srz_zumix_ reported at:  http://trac.cppcheck.net/ticket/7782 :  

Cppcheck v1.76.1
Wrong path will be output to the xml "file" attribute

For example:

current directory: C:\Temp
realFilename("C:\Temp\main.cpp") == "Temp\Temp\main.cpp"

current directory: C:\
realFilename("C:\Temp\main.cpp") == "C:\Temp\main.cpp"

More details:

FindFirstFile?
​https://msdn.microsoft.com/en-us/library/aa364418(v=vs.85).aspx

```
As stated previously, you cannot use a trailing backslash (\) in the lpFileName input string for FindFirstFile?, therefore it may not be obvious how to search root directories.
```

current directory: C:\
FindFirstFile?("C:", &fd) == INVALID_HANDLE_VALUE

current directory: C:\Temp
FindFirstFile?("C:", &fd) != INVALID_HANDLE_VALUE
fd.cFileName is "Temp"

current directory: D:\
FindFirstFile?("C:", &fd) == INVALID_HANDLE_VALUE

**Comment by amai2012** [comment]
That shortcut seems to improve things:
~~~~C++
static bool realFileName(const std::vector<TCHAR> &buf, std::ostream &ostr) {
	// Detect root directory, see simplecpp:realFileName returns the wrong root path #45
	if ((buf.size()==2 || (buf.size()>2 && buf[2]=='\0'))
		&& std::isalpha(buf[0]) && buf[1]==':')
	{
		ostr << (char)buf[0];
		ostr << (char)buf[1];
		return true;
	}
    WIN32_FIND_DATA FindFileData;
    HANDLE hFind = FindFirstFile(&buf[0], &FindFileData);
    if (hFind == INVALID_HANDLE_VALUE)
        return false;
    for (const TCHAR *c = FindFileData.cFileName; *c; c++)
        ostr << (char)*c;
    FindClose(hFind);
    return true;
}
~~~~

**Comment by amai2012** [comment]
Shoud be fixed by #55 (sry for misleading name)
Test code:
~~~~C++
/*! \return true on success, false else */
bool test(const char* fileName, const char* currentDirecory)
{
	_mkdir(currentDirecory);
	_chdir(currentDirecory);
	{
		char buf[1024]={0};
		_getcwd(buf, 1024);
		assert(strcmp(currentDirecory, buf)==0);
	}
        FILE* f=fopen(fileName, "w");
        fputs(" ", f);
	fclose(f);
	std::string result = realFilename(fileName);
	std::replace( result.begin(), result.end(), '/', '\\');
	remove(fileName);
	return result == fileName;
}

int main(int argc, char **argv) {
	int failures=0;
	if (!test("C:\\Temp\\Main.cpp", "C:\\")) ++failures;
	if (!test("C:\\Temp\\Main.cpp", "C:\\Temp")) ++failures;
	if (!test("C:\\Temp\\Main.cpp", "C:\\Windows")) ++failures;
	if (!test("C:\\Temp\\Main.cpp", "D:\\")) ++failures;

	exit(failures);
}
~~~~
